src/ngx_http_lua_headers_in.c

```diff
@@ -645,14 +645,95 @@ ngx_http_clear_builtin_header(ngx_http_request_t *r,
 }
 
 
+static ngx_hash_init_t  lua_req_header_hash;
+static u_char           header_string[512];
```

Comment:
Again: let's avoid these global variables. Please see how ngx_http_proxy_module in the nginx core initializes its "proxy_headers_hash".